### PR TITLE
Feature/blui 4130 verify code screen base

### PR DESCRIPTION
**Pull request by JeffGreiner-eaton**
Fixes BLUI-4130

#### Changes proposed in this Pull Request:

- Add Verify code screen base
-
-

#### Screenshots / Screen Recording (if applicable)

image
image

#### To Test:

- yarn start:example2
- select test verify code screen
- open dev tools> console verify calls on validator and resend

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        showNext: otherProps.showNext,
+        previousLabel: otherProps.previousLabel,
+        nextLabel: otherProps.nextLabel,
+        onPrevious: otherProps.onPrevious,
+        onNext: otherProps.onNext,
+        currentStep: otherProps.currentStep,
+        totalSteps: otherProps.totalSteps,
+        fullWidthButton: otherProps.fullWidthButton,
+        // @TODO: should we extend the rest of the props (CardActionsProps) or should we set this up to take in each sections props separately e.g., workflowCardProps, actionsProps, etc.
+    };
```
Comment (daileytj) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:59]: @joebochill Any thoughts on the @TODO here? Would be nice to be able to pass the full prop objects to the card, header, instructions, etc.

**Review by daileytj — COMMENTED** [review]
Aside from the `actionProps` @TODO I left in there, I think this is good to go. The defaults and translations will be handled by the full component. Great job creating an example to review in the example2 project as well. Going to hold my approval until at least 1 other person looks at this.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * @param codeValidator used to test the input for valid formatting
+ * @param onResend function to call when the user clicks the 'resend code' button
+ * @param resendInstructions text to display ahead of the resend link/button
+ * @param resendLabel label for the resend link/button
+ * @param initialValue code used to pre-populate the field
+ *
+ * @category Component
+ */
+
+export const VerifyCodeScreenBase: React.FC<React.PropsWithChildren<React.PropsWithChildren<VerifyCodeScreenProps>>> = (
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:25]: There is no need of 2 React.PropsWithChildren. One React.PropsWithChildren should work.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    props
+) => {
+    const {
+        codeValidator = (code: string): boolean | string => {
+            if (code?.length > 0) {
+                return true;
+            }
+            return 'You must provide a valid code';
+        },
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:34]: This can be simplified as:
codeValidator = (code: string): boolean | string => (code?.length > 0 ? true : 'You must provide a valid code'),

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            if (typeof validatorResponse === 'boolean' && validatorResponse === true) {
+                setIsCodeValid(true);
+                setCodeError('');
+            } else if (typeof validatorResponse === 'boolean' && validatorResponse === false) {
+                setIsCodeValid(false);
+                setCodeError('');
+            } else if (typeof validatorResponse === 'string') {
+                setIsCodeValid(false);
+                setCodeError(validatorResponse);
+            }
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:80]: This block can be simplified to:
```
setIsCodeValid(typeof validatorResponse === 'boolean' ? validatorResponse : false);
setCodeError(typeof validatorResponse === 'string' ? validatorResponse : '');
```

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        },
+        [codeValidator, verifyCode]
+    );
+
+    return (
+        <WorkflowCard>
+            <WorkflowCardHeader title={title}></WorkflowCardHeader>
+            <WorkflowCardBody>
+                <WorkflowCardInstructions divider instructions={instructions}></WorkflowCardInstructions>
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:89]: As there are no children passed to WorkflowCardInstructions, we can use self closing tag:
`<WorkflowCardInstructions divider instructions={instructions} />`

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/types.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { WorkflowCardProps } from '../../components/WorkflowCard/WorkflowCard.types';
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/types.ts:1]: This is not related to this PR. @manojleaton We should have moved individual type definitions of WorkflowCard, WorkflowCardActions, WorkflowCardBody, WorkflowCardHeader, WorkflowCardInstructions to their respective files/folders and export them using index.ts.

@joebochill @daileytj

**Review by surajeaton — COMMENTED** [review]
Please resolve the merge conflicts.

**Comment by JeffGreiner-eaton** [comment]
@daileytj @surajeaton this is ready for another round of review

**Review by daileytj — APPROVED** [review]
